sklearn/impute/_base.py

```diff
@@ -913,6 +946,28 @@ def fit_transform(self, X, y=None):
 
         return imputer_mask
 
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Input features.
+
+            - If `input_features` is `None`, then `feature_names_in_` is
+              used as feature names in. If `feature_names_in_` is not defined,
+              then names are generated: `[x0, x1, ..., x(n_features_in_)]`.
+            - If `input_features` is an array-like, then `input_features` must
+              match `feature_names_in_` if `feature_names_in_` is defined.
+
+        Returns
+        -------
+        feature_names_out : ndarray of str objects
+            Transformed feature names.
+        """
+        input_features = _check_feature_names_in(self, input_features)
+        return input_features[self.features_]
```

Comment:
I do not think we have a rule. I went with `SimpleImputer` prefixing `"indicator_"` because it needed to distinguish between the imputed features and the indicator.

Looking at this again, I think we can go with `MissingIndcator` to add the prefix. This way `SimpleImputer` only needs to combine them.